frontend/controller/actions/group.js

```diff
@@ -238,30 +235,27 @@ export default (sbp('sbp/selectors/register', {
       const contractID = message.contractID()
 
       // After the contract has been created, store pesistent keys
-      sbp('chelonia/storeSecretKeys',
-        () => [CEK, CSK, inviteKey].map(key => ({ key }))
+      await sbp('chelonia/storeSecretKeys',
+        new Secret([CEK, CSK, inviteKey].map(key => ({ key })))
       )
 
-      await sbp('chelonia/queueInvocation', contractID, ['gi.actions/identity/joinGroup', {
-        contractID: userID,
-        data: {
-          groupContractID: contractID,
-          inviteSecret: serializeKey(CSK, true),
-          creatorID: true
-        }
-      }])
+      await sbp('chelonia/contract/wait', contractID).then(() => {
+        return sbp('gi.actions/identity/joinGroup', {
+          contractID: userID,
+          data: {
+            groupContractID: contractID,
+            inviteSecret: serializeKey(CSK, true),
+            creatorID: true
+          }
+        })
+      })
 
-      return message
+      return message.contractID()
```

Comment:
Why was this change made?

Reply:
To avoid having to copy GIMessage across boundaries (a potentially expensive operation)